Numpy 2 compatibility issue: nilearn/interfaces/fmriprep/load_confounds

https://github.com/nilearn/nilearn/actions/runs/7826655703/job/21352926865#step:6:20258

```
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy0-param0-1.4.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy0-param0-21.x.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy1-param1-1.4.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy1-param1-21.x.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy2-param2-1.4.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy2-param2-21.x.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy3-param3-1.4.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy3-param3-21.x.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy4-param4-1.4.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy4-param4-21.x.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy5-param5-1.4.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy5-param5-21.x.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy6-param6-1.4.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy6-param6-21.x.x]
FAILED nilearn/interfaces/fmriprep/tests/test_load_confounds.py::test_nilearn_regress[test_strategy7-param7-1.4.x]
```


```
     def _handle_non_steady(confounds):
        """Simulate non steady state correctly while increase the length."""
        X = confounds.values
        # the first row is non-steady state, replace it with the input from the
        # second row
        non_steady = X[0, :]
>       X[0, :] = X[1, :]
E       ValueError: assignment destination is read-only
```

--- a/.github/workflows/numpy2_compatibility.yml
+++ b/.github/workflows/numpy2_compatibility.yml
@@ -4,6 +4,12 @@
 name: Numpy2 compatibility
 
 on:
+
+    # Comment before merging in main
+    pull_request:
+        branches:
+        -   '*'
+
     schedule:
     # Run every day at 8am UTC
     -   cron: 0 8 * * *

--- a/.github/workflows/numpy2_compatibility.yml
+++ b/.github/workflows/numpy2_compatibility.yml
@@ -6,9 +6,9 @@ name: Numpy2 compatibility
 on:
 
     # Comment before merging in main
-    pull_request:
-        branches:
-        -   '*'
+    # pull_request:
+    #    branches:
+    #    -   '*'
 
     schedule:
     # Run every day at 8am UTC

--- a/nilearn/interfaces/fmriprep/tests/test_load_confounds.py
+++ b/nilearn/interfaces/fmriprep/tests/test_load_confounds.py
@@ -84,19 +84,23 @@ def _simu_img(tmp_path, demean):
 
 
 def _handle_non_steady(confounds):
-    """Simulate non steady state correctly while increase the length."""
+    """Simulate non steady state correctly while increase the length.
+
+    - The first row is non-steady state,
+      replace it with the input from the second row.
+
+    - Repeat X in length (axis = 0) 10 times to increase
+      the degree of freedom for numerical stability.
+
+    - Put non-steady state volume back at the first sample.
+    """
     X = confounds.values
-    # the first row is non-steady state, replace it with the input from the
-    # second row
     non_steady = X[0, :]
-    X[0, :] = X[1, :]
-    # repeat X in length (axis = 0) 10 times to increase
-    # the degree of freedom for numerical stability
-    X = np.tile(X, (10, 1))
-    # put non-steady state volume back at the first sample
-    X[0, :] = non_steady
-    X = pd.DataFrame(X, columns=confounds.columns)
-    return X
+    tmp = np.vstack((X[1, :], X[1:, :]))
+    tmp = np.tile(tmp, (10, 1))
+    return pd.DataFrame(
+        np.vstack((non_steady, tmp[1:, :])), columns=confounds.columns
+    )
 
 
 def _regression(confounds, tmp_path):

